esmvaltool/diag_scripts/sea_surface_salinity/compare_salinity.py

```diff
@@ -0,0 +1,261 @@
+import logging
+import string
+import os
+from datetime import datetime
+from functools import reduce
+import matplotlib.pyplot as plt
+from matplotlib.legend_handler import HandlerBase
+from matplotlib.text import Text
+from matplotlib.legend import Legend
+import numpy as np
+import cf_units
+import iris
+from iris.util import unify_time_units
+import iris.quickplot as qplot
+
+from iris.analysis.stats import pearsonr
+from iris.coord_categorisation import add_month_number, add_year
+
+import esmvaltool.diag_scripts.shared
+import esmvaltool.diag_scripts.shared.names as n
+from esmvaltool.diag_scripts.shared import group_metadata
+from esmvaltool.diag_scripts.shared._base import ProvenanceLogger
+
+from esmvalcore.preprocessor._time import regrid_time
+
+
+logger = logging.getLogger(__name__)
+
+
+class CompareSalinity(object):
+    def __init__(self, config):
+        self.cfg = config
+
+    def compute(self):
+        data = group_metadata(self.cfg[n.INPUT_DATA].values(), n.SHORT_NAME)
+        for short_name in data:
+            logger.info("Processing variable %s", short_name)
+            variables = group_metadata(data[short_name], n.ALIAS)
+            ref_alias = list(variables.values())[0][0]['reference_dataset']
+            reference_dataset = variables.pop(ref_alias)[0]
+            reference = iris.load_cube(reference_dataset[n.FILENAME])
+            reference_ancestor = reference_dataset[n.FILENAME]
+            logger.debug(reference)
+            for alias, dataset_info in variables.items():
+                logger.info("Plotting dataset %s", alias)
+                dataset_info = dataset_info[0]
+                dataset = iris.load_cube(dataset_info[n.FILENAME])
+                time_coord = dataset.coord('time')
+                if time_coord.units.calendar == 'proleptic_gregorian':
+                    time_coord.units = cf_units.Unit(
+                        time_coord.units.name,
+                        calendar='gregorian',
+                    )
+                unify_time_units((reference, dataset))
```

Comment:
will this always succeed? Probably best to catch an error with a `try/except` statement

Reply:
Not sure why @jvegasbsc implemented this. I would leave it as is since it seems to be working fine...

Reply:
sure, not a biggie